python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -85,17 +85,21 @@ def run(self, dia_sources, exposure_time):
             (`pandas.DataFrame`)
         """
 
-        flag_map = os.path.join(utils.getPackageDir("ap_association"), "data/association-flag-map.yaml")
-        unpacker = UnpackApdbFlags(flag_map, "DiaSource")
-        flags = unpacker.unpack(dia_sources["flags"], "flags")
+        if "flags" in dia_sources.columns:
+            flag_map = os.path.join(utils.getPackageDir("ap_association"), "data/association-flag-map.yaml")
+            unpacker = UnpackApdbFlags(flag_map, "DiaSource")
+            flags = unpacker.unpack(dia_sources["flags"], "flags")
+            trail_edge_flags = flags["ext_trailedSources_Naive_flag_edge"]
+        else:
+            trail_edge_flags = dia_sources["trail_flag_edge"]
```

Comment:
I'm curious why you need the `if` here? We're removing `flags` everywhere, so where could it be coming from? The commit message doesn't say either.

Reply:
My perspective is that we are moving to unpacking the bitpacked flags by default, not removing the capability to do so entirely (until after the deprecation period).  That requires a bit of extra code but isn't a substantial increase in complexity.